### PR TITLE
remove unnecessary empty `build.gradle` files

### PR DESCRIPTION
**Pull request by rursprung**
### Description
these have no impact, as the targets are added by the parent `build.gradle`. thus they do not need to exist and are currently just cluttering the workspace and confusing people trying to work in that area.

see also the discussion on https://github.com/opensearch-project/opensearch-build/issues/99 for some details.

Signed-off-by: Ralph Ursprung <Ralph.Ursprung@avaloq.com>

### Issues Resolved
n/a

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by rursprung** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> 
>     * **RESULT:** FAILURE ❌
> 
>     * **URL:** https://build.ci.opensearch.org/job/gradle-check/4242/
> 
>     * **CommitID:** [17170ec](https://github.com/opensearch-project/OpenSearch/commit/17170ec11c12bc365c930501e0215873777e5eec)

this _might_ be a relevant failure:
```
FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':distribution:docker:docker-export:exportDockerImage'.
> A problem occurred starting process 'command 'docker''
```
however, i sadly currently don't have a linux box with docker on it (and a connection to the internet) available to test this... could somebody else maybe briefly try to run this with `--info` to see more details? thanks!

**Comment by dblock** [comment]
I just kicked it to see if it's transient. 

Does `./gradle :distribution:docker:docker-export:exportDockerImage` fail the same way for you locally?

**Comment by rursprung** [comment]
> Does `./gradle :distribution:docker:docker-export:exportDockerImage` fail the same way for you locally?

i wouldn't know, our infrastructure doesn't support having a local docker installation 😞  (and the OpenSearch gradle build doesn't support using Docker CLI with a remote docker server)

**Comment by rursprung** [comment]
> > Does `./gradle :distribution:docker:docker-export:exportDockerImage` fail the same way for you locally?
> 
> i wouldn't know, our infrastructure doesn't support having a local docker installation 😞 (and the OpenSearch gradle build doesn't support using Docker CLI with a remote docker server)

i've now temporarily patched the gradle build to use Docker CLI, then it works. i've also raised #4883 to get a more long-term solution for this.

i can confirm that the gradle target builds just fine for me locally. so i've now rebased and pushed that again in the hopes that the jenkins build will now run through.

**Comment by rursprung** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> 
>     * **RESULT:** FAILURE ❌
> 
>     * **URL:** https://build.ci.opensearch.org/job/gradle-check/5337/
> 
>     * **CommitID:** [695b01f](https://github.com/opensearch-project/OpenSearch/commit/695b01f40eb7e8ca2ec728511364c6f3cb462391)

see my comment above: in my environment this gradle target runs just fine, so this might be linux-specific. if somebody with access to a linux box (where they can build this project) could test if they can reproduce this error that'd be great, thanks! otherwise i'm a bit stuck here.

however, this is also not an important PR, it's just a minor cleanup...

**Comment by rursprung** [comment]
rebased in the hopes that this can now be merged (previously it failed with some unrelated false-rejects...)

**Comment by rursprung** [comment]
this failure seems to be a false reject, i can run the specified gradle target locally with no issues:
```
* What went wrong:
Execution failed for task ':distribution:docker:docker-export:exportDockerImage'.
> A problem occurred starting process 'command 'docker''
```

**Comment by rursprung** [comment]
> @rursprung: Are you still working on this PR ?

not sure what i can do here if it always hits false rejects... (i don't think that there was a single PR i tried to contribute to the OpenSearch repo which didn't get hit with several false rejects before passing through CI - it's not exactly encouraging to contribute this way 😞).
i've rebased the PR once more, let's see whether that helps... 🤷

**Review by dreamer-89 — APPROVED** [review]
LGTM ! Thank you @rursprung for fixing this. Appreciate the effort and perseverance in fixing this. 👍

**Comment by andrross** [comment]
> not sure what i can do here if it always hits false rejects... (i don't think that there was a single PR i tried to contribute to the OpenSearch repo which didn't get hit with several false rejects before passing through CI - it's not exactly encouraging to contribute this way 😞).

@rursprung Completely agree this is not good! We have been making steady improvements to the tests and CI infrastructure to make this more reliable, so hopefully things have improved since this PR was initially opened.

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Remove LegacyESVersion.V_7_6_ and V_7_7_ Constants ([#4837](https://github.com/opensearch-project/OpenSearch/pull/4837))
 - Remove LegacyESVersion.V_7_10_ Constants ([#5018](https://github.com/opensearch-project/OpenSearch/pull/5018))
 - Remove Version.V_1_ Constants ([#5021](https://github.com/opensearch-project/OpenSearch/pull/5021))
+- Removed unnecessary empty `build.gradle` files ([#4723](https://github.com/opensearch-project/OpenSearch/pull/4723))
```
Comment (andrross) [inline CHANGELOG.md:69]: We can omit the changelog entry for build related changes like this. See https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#changelog

Comment (rursprung) [reply CHANGELOG.md]: ah, that rule changed in the meantime :)
i've removed it now (& also used the chance to rebase again, because why not)

**Comment by rursprung** [comment]
well, removing that changelog entry (& rebasing) was a mistake... we're back to a random build failure 😠

**Comment by tlfeng** [comment]
@rursprung No worries, we can help you re-run the failed build job.

**Comment by andrross** [comment]
This seems to reliably fail with:

```
* What went wrong:
Execution failed for task ':distribution:docker:docker-export:exportDockerImage'.
> A problem occurred starting process 'command 'docker''
```

I have no idea why this PR would cause that, but it does not seem to be flakiness.

**Comment by kartg** [comment]
@rursprung this will need a rebase to get past the following Gradle check failure, to determine if `exportDockerImage` is still failing:

```
* What went wrong:
Execution failed for task ':distribution:bwc:minor:buildBwcLinuxTar'.
> Building 2.8.0 didn't generate expected file /var/jenkins/workspace/gradle-check/search/distribution/bwc/minor/build/bwc/checkout-2.x/distribution/archives/linux-tar/build/distributions/opensearch-min-2.8.0-SNAPSHOT-linux-x64.tar.gz
```

**Comment by kartg** [comment]
I'm able to reproduce this failure locally.

```
$ ./gradlew --stacktrace :distribution:docker:docker-export:exportDockerImage
...
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':distribution:docker:docker-export:exportDockerImage'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:149)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:282)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:147)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:135)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:74)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:337)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:324)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:317)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:303)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:463)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:380)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
Caused by: org.gradle.process.internal.ExecException: A problem occurred starting process 'command 'docker''
        at org.gradle.process.internal.DefaultExecHandle.execExceptionFor(DefaultExecHandle.java:241)
        at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:218)
        at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:370)
        at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:87)
        at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:49)
Caused by: net.rubygrapefruit.platform.NativeException: Could not start 'docker'
        at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:27)
        at net.rubygrapefruit.platform.internal.WrapperProcessLauncher.start(WrapperProcessLauncher.java:36)
        at org.gradle.process.internal.ExecHandleRunner.startProcess(ExecHandleRunner.java:98)
        at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:71)
        ... 3 more
Caused by: java.io.IOException: Cannot run program "docker" (in directory "<redacted-OS-root-path>/distribution/docker/docker-export"): error=2, No such file or directory
        at net.rubygrapefruit.platform.internal.DefaultProcessLauncher.start(DefaultProcessLauncher.java:25)
        ... 6 more
Caused by: java.io.IOException: error=2, No such file or directory
        ... 7 more
```

**Comment by kartg** [comment]
Through empirical testing, I've narrowed down the failure to be related to the removal of the `distribution/docker/docker-export/build.gradle` file. I'm going to push an additional commit to restore the file and verify that Gradle check succeeds. I'm not sure why/how the absence of this file causes the failure, so a deeper dive is probably required before we can merge this in.

**Comment by kartg** [comment]
```
org.opensearch.cluster.remote.test.RemoteClustersIT > testHAProxyModeConnectionWorks FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([77F537DC32B80579:7083E217093E7324]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks(RemoteClustersIT.java:129)

```

https://github.com/opensearch-project/OpenSearch/issues/1703, will refire

**Comment by kartg** [comment]
Gradle check passes (the failure above is a [known flaky test failure](https://github.com/opensearch-project/OpenSearch/issues/5957)), and this change is ready to merge. Flipping to draft so we can dig deeper into why the `distribution/docker/docker-export/build.gradle` file [is required](https://github.com/opensearch-project/OpenSearch/pull/4723#issuecomment-1593752965).

**Comment by kartg** [comment]
Flipping out of draft so this can be merged. I've opened https://github.com/opensearch-project/OpenSearch/issues/8112 to track the investigation/removal of the `docker-export/build.gradle` file.

**Pull request merged**